yt/frontends/amrvac/datfile_utils.py

```diff
@@ -151,17 +151,12 @@ def get_single_block_data(istream, byte_offset, block_shape):
 
 def get_single_block_field_data(istream, byte_offset, block_shape, field_idx):
     """retrieve a specific block (ONE field) from a datfile"""
-    istream.seek(byte_offset)
-
     # compute byte size of a single field
     field_shape = block_shape[:-1]
     fmt = ALIGN + np.prod(field_shape) * "d"
     byte_size_field = struct.calcsize(fmt)
 
-    # Read actual data
-    istream.seek(byte_size_field * field_idx, 1)  # seek forward
-    d = struct.unpack(fmt, istream.read(struct.calcsize(fmt)))
-
-    # Fortran ordering
-    block_field_data = np.reshape(d, field_shape, order="F")
-    return block_field_data
+    istream.seek(byte_offset + byte_size_field * field_idx)
+    data = np.fromfile(istream, "=f8", count=np.prod(field_shape))
+    data.shape = field_shape[::-1]
```

Comment:
One trick we've used for hdf5 files has been to create a destination array and read directly into that.  To be honest this did make me stop and think what I was seeing.

Reply:
I'm also not super happy with how this looks. `[::-1]` may be a common(ish) idiom but it doesn't feel right to me.
I don't think `numpy.fromfile` has a "dest" (or similar) argument, where should I look for the technique you're mentioning ?

Reply:
I also couldn't find a good one.  All I could come up with was doing some kind of `destination_buffer[;] = np.fromfile` but that's not really very helpful.  Let me keep looking.  I think one possibility would be to do a `view` on the results of `np.fromfile`.